app/src/main/java/com/douglasbruce/tasky/core/data/repository/EventRepositoryImpl.kt

```diff
@@ -46,13 +47,15 @@ class EventRepositoryImpl @Inject constructor(
             serializer.toJson(event.toCreateEventRequest(), CreateEventRequest::class.java)
 
         return authenticatedRetrofitCall(serializer) {
-            taskyNetwork.createEvent(
+            val networkEvent = taskyNetwork.createEvent(
                 createEventRequest = MultipartBody.Part.createFormData(
                     "create_event_request",
                     requestJson!!
                 ),
                 photos = emptyList()
             )
+
+            dao.upsertEvent(networkEvent.toEvent().toEventEntity())
```

Comment:
What would happen if the network call succeeded, but the coroutine was cancelled during this `upsertEvent` call (for example by navigating back, so `viewModelScope` is destroyed)? Then, you could potentially have an inconsistent state, since the local result does not reflect the remote one.
To avoid this, you can do this:
```
withContext(NonCancellable) {
    dao.upsertEvent(...)
}
```